config/base.js

```diff
@@ -37,11 +38,16 @@ const SCHEMA_CHILD_PIPELINES = Joi.object()
         startAll: Joi.boolean()
     })
     .unknown(false);
-
+const SCHEMA_SETUP_TEARDOWN_JOB = Job.job.keys({ requires: Joi.forbidden() });
 const SCHEMA_STAGE = Joi.object()
     .keys({
         description: Joi.string(),
-        jobs: Joi.array().items(Job.jobName).min(0)
+        jobs: Joi.array().items(Job.jobName).min(0),
+        setup: SCHEMA_SETUP_TEARDOWN_JOB,
+        teardown: SCHEMA_SETUP_TEARDOWN_JOB,
+        requires: Job.requires,
+        startFrom: Job.jobName,
+        workflowGraph: WorkflowGraph.workflowGraph
```

Comment:
Will stage `workflowGraph` contain edges from `setup` to jobs with in the stage?

Reply:
we can add the setup edges but teardown is trickier